8262099: jcmd VM.metaspace should report unlimited size if MaxMetaspaceSize isn't specified

kelthuzadx opened the pull request:
Hi,

Can I have a review of this minor fix about `jcmd VM.metaspace`?

When MaxMetaspaceSize is not specified, it would set to max_uintx, jcmd VM.metaspace reports metaspace summary information as follows:

$ ./jcmd <pid> VM.metaspace | grep MaxMetaspaceSize
MaxMetaspaceSize: 17179869184.00 GB

Actually, this is a bug, the expected output should be:

MaxMetaspaceSize: unlimited

The root cause of the problem is that the MaxMetaspaceSize rounding performed during arguments parsing is inconsistent with the round checking in print_settings(metaspaceRerporter.cpp).

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262099](https://bugs.openjdk.java.net/browse/JDK-8262099): jcmd VM.metaspace should report unlimited size if MaxMetaspaceSize isn't specified


### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to ca16d9cc18b3adf00f14ea535fdb0f1da523d84e
 * [Lutz Schmidt](https://openjdk.java.net/census#lucy) (@RealLucy - **Reviewer**)


### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2671/head:pull/2671`
`$ git checkout pull/2671`

dholmes-ora commented on src/hotspot/share/memory/metaspace/metaspaceReporter.cpp:
```diff
@@ -1,6 +1,6 @@
- * Copyright (c) 2018, 2020, Oracle and/or its affiliates. All rights reserved.
- * Copyright (c) 2018, 2020 SAP SE. All rights reserved.
+ * Copyright (c) 2018, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, 2021 SAP SE. All rights reserved.
```
Please note that while Oracle requests/requires that it's copyright line gets updated each year a file is modified, it is not clear whether that extends to any other copyright lines that might appear in the file. In general for companies other than Oracle, only a committer from that company should update a copyright line for that company.

Thanks,
David

kelthuzadx replied:
Changed. Thanks for pointing out this.

tstuefe replied:
I would have been fine with the copyright change :)

tstuefe reviewed (approved):
Change looks fine, thanks for fixing. Can you please add a comment like "see Metaspace::ergo_initialize()" ?

I think the only reason we align MaxMetaspaceSize down is to forestall any comparison overflows on metaspace expansion. But I am not sure this is even an issue. I would like a solution where we leave it at max_uint and check the comparisons for overflow instead. But this patch looks fine.

Make sure you run the associated JTreg test.

Cheers, Thomas

kelthuzadx commented:
Hi Thomas, I've updated the patch according to your review comment. All related tests have passed. In addition, I added a new test case to ensure that it works well in the future. Also I have updated the SAP copyright year.(Please let me know if I misunderstand your meaning:-))

Thanks,
Yang Yi

kelthuzadx commented:
/integrate

RealLucy reviewed (approved):
Changes look good to me. 
Thanks for fixing. I was complaining about the, well, awful printout at another place.

kelthuzadx commented:
Thanks @dholmes-ora @tstuefe @RealLucy for reviews. Since I'm not a committer I can not commit directly, can you help to sponsor this patch :) ? 

Thanks,
Yang

dholmes-ora commented:
/sponsor
